lib/server.js

```diff
@@ -98,6 +100,19 @@ function initSync(mediator, mbaasApi, datasetId, syncOptions) {
   return deferred.promise;
 }
 
+/**
+ * If the objects _id field is a Mongodb~ObjectID, convert it to a string.
+ *
+ * @param {String|Object} object - The object to check.
+ * @returns {String} - The original _id field or the converted Mongodb~ObjectID
+ */
+function getObjectId(object) {
```

Comment:
@aidenkeating  @nialldonnellyfh I think having this logic in raincatcher-sync makes sense.
Sync is only concerned with synchronisation of JSON data. An ObjectID is not valid JSON, so there will be unexpected behaviour.